### PR TITLE
Preview Website

### PR DESCRIPTION
**Pull request by Patrick-Kladek**
### Description

Add command to generate a nice preview website with all images.

### Tasks

- [x] Add HTML Framework - SwiftSoup
- [x] Create Website based exported files.

### Infos For Reviewer

Inspired by fastlane's `screenshots.html` file. Do we need to add attribution?

### Finished Result
Bildschirmfoto 2021-12-07 um 13 57 38

### CODE REVIEW

File: Screenshot Framer.xcodeproj/xcshareddata/xcschemes/Screenshot-Framer-CLI.xcscheme [hunk 1]
```diff
@@ -59,6 +59,16 @@
             ReferencedContainer = "container:Screenshot Framer.xcodeproj">
          </BuildableReference>
       </BuildableProductRunnable>
+      <CommandLineArguments>
+         <CommandLineArgument
+            argument = "export --input /Users/patrick/Developer/Bikemap-AppStore/Input"
+            isEnabled = "NO">
+         </CommandLineArgument>
+         <CommandLineArgument
+            argument = "website --export-folder /Users/patrick/Developer/Bikemap-AppStore/export"
```
Comment (schwmi) [inline Screenshot Framer.xcodeproj/xcshareddata/xcschemes/Screenshot-Framer-CLI.xcscheme:68]: nit: would remove those hardcoded paths

File: Screenshot Framer CLI/ImagesParser.swift [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    struct Screen {
+        let groups: [Group]
+        let name: String
+    }
+
+    private let fileManager = FileManager()
+
+    // MARK: - ImagesParser
+
+    func languages(in folder: URL) throws -> [Language] {
```
Comment (schwmi) [inline Screenshot Framer CLI/ImagesParser.swift:41]: It is not fully obvious why the `ImagesParser` offers a method to read languages from a URL. Maybe a clarifying comment on top of the class helps here.

File: Screenshot Framer CLI/ImagesParser.swift [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+            self.language = url.pathComponents.dropLast().last!
+        }
+
+        var debugDescription: String {
+            return "\(self.device) \(self.number)"
+        }
+    }
+
+    func contents(in folder: URL) throws -> Language {
```
Comment (schwmi) [inline Screenshot Framer CLI/ImagesParser.swift:104]: maybe a better name would be here `language(in folder: URL)`

File: Screenshot Framer CLI/Export.swift [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            ConsoleIO.writeMessage("Here are the error codes:", to: .error)
+            ConsoleIO.writeMessage("\(exportErrors.map { $0.rawValue }.joined(separator: "\n"))")
+        }
+    }
+
+    func checkedErrors(_ errors: [LayoutError]) -> [LayoutError] {
+        var exportErrors = errors
+
+        if self.ignoreWarnings {
+            exportErrors.remove(object: .fontToBig)
```
Comment (schwmi) [inline Screenshot Framer CLI/Export.swift:92]: nit: guess this should be called `.fontTooBig`

**Review by schwmi — APPROVED** [review]
Very nice addition 👍

File: Screenshot Framer CLI/Export.swift [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+        var isDir: ObjCBool = false
+        fileManager.fileExists(atPath: self.input.path, isDirectory: &isDir)
+
+        let projects: [URL]
+        if isDir.boolValue {
+            let contentOfDirectory = try fileManager.contentsOfDirectory(at: self.input, includingPropertiesForKeys: nil, options: [.skipsHiddenFiles, .skipsPackageDescendants, .skipsSubdirectoryDescendants])
+            projects = contentOfDirectory.filter { $0.pathExtension == "frame" }
+        } else {
+            projects = [self.input]
```
[the remark this answers is not in the document]
Comment (Patrick-Kladek) [reply Screenshot Framer CLI/Export.swift:33]: Good idea 👍

**Pull request merged**